source/Engine.cpp

```diff
@@ -354,16 +354,18 @@ void Engine::Place()
 		// taking off from a specific planet or nearby.
 		if(ship->GetSystem() == system && !ship->IsDisabled())
 		{
-			const Personality &person = ship->GetPersonality();
+			const Personality &personality = ship->GetPersonality();
 			bool hasOwnPlanet = ship->GetPlanet();
-			bool launchesWithPlayer = (planet && planet->CanLand(*ship))
-					&& !person.IsStaying() && !person.IsWaiting()
-					&& (!hasOwnPlanet || (ship->IsYours() && ship->GetPlanet() == planet));
+			bool launchesWithPlayer = (ship->GetPlanet() == planet
+					&& !personality.IsStaying() && !personality.IsWaiting());
 			const StellarObject *object = hasOwnPlanet ?
 					ship->GetSystem()->FindStellar(ship->GetPlanet()) : nullptr;
 			// Default to the player's planet in the case of data definition errors.
-			if(person.IsLaunching() || launchesWithPlayer || (hasOwnPlanet && !object))
+			if(personality.IsLaunching() || launchesWithPlayer || (hasOwnPlanet && !object))
 			{
+				if(hasOwnPlanet)
```

Comment:
Isn't this going to trigger for the other cases as well, say for a player escort that landed with the player on a planet?